apps/src/applab/designElements/photochooser.jsx

```diff
@@ -0,0 +1,226 @@
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+import React from 'react';
+import PropertyRow from './PropertyRow';
+import BooleanPropertyRow from './BooleanPropertyRow';
+import ColorPickerPropertyRow from './ColorPickerPropertyRow';
+import ZOrderRow from './ZOrderRow';
+import EventHeaderRow from './EventHeaderRow';
+import EventRow from './EventRow';
+import BorderProperties from './BorderProperties';
+import themeValues from '../themeValues';
+import {ICON_PREFIX_REGEX} from '../constants';
+import * as elementUtils from './elementUtils';
+import * as assetPrefix from '../../assetManagement/assetPrefix';
+import designMode from '../designMode';
+import elementLibrary from './library';
+
+class PhotoChooserProperties extends React.Component {
+  static propTypes = {
+    element: PropTypes.instanceOf(HTMLElement).isRequired,
+    handleChange: PropTypes.func.isRequired,
+    onDepthChange: PropTypes.func.isRequired
+  };
+
+  handleIconColorChange = value => {
+    this.props.handleChange('icon-color', value);
+    this.props.handleChange(
+      'image',
+      this.props.element.getAttribute('data-canonical-image-url')
+    );
+  };
+
+  render() {
+    const element = this.props.element;
+
+    let iconColorPicker;
+    const canonicalImage = element.getAttribute('data-canonical-image-url');
+    if (ICON_PREFIX_REGEX.test(canonicalImage)) {
+      iconColorPicker = (
+        <ColorPickerPropertyRow
+          desc={'icon color'}
+          initialValue={elementUtils.rgb2hex(
+            element.getAttribute('data-icon-color') || '#000000'
+          )}
+          handleChange={this.handleIconColorChange}
+        />
+      );
+    }
+
+    return (
+      <div id="propertyRowContainer">
+        <PropertyRow
+          desc={'id'}
+          initialValue={elementUtils.getId(element)}
+          handleChange={this.props.handleChange.bind(this, 'id')}
```

Comment:
Rather than using bind everywhere, we should use arrow notation (other designElements used the older bind notation, and we should start migrating to the new notation)
i.e.
`() => this.props.handleChange('id')`

Reply:
we paired on this and couldn't get the arrow notation to play nice so sticking with bind